Coding Style Guidline.rst

```diff
@@ -0,0 +1,285 @@
+    
+    if i == 0
+        if x == 3
+            x = 10
+
+**Bad:**
+
+::   
+  
+  if i == 0
+    if x == 3
+     x = 10
+
+2.3 Whitespaces Around Arithmetic Operators
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There must be whitespaces surrounding arithmetic operators
+
+**Good:**
+
+::
+
+   x = 3 + 5 / 2
+
+**Bad:**
+
+::
+
+   x=3+5/2
+
+2.4 Missing White Spaces After Commas
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There must be a whitespace following a comma
+
+**Good:**
+
+::
+
+   myFunctionCall('a', 'b', 'c')
+
+**Bad:**
+
+::
+
+   myFunctionCall('a','b','c')
+
+2.5 Header File
+~~~~~~~~~~~~~~~
+
+Every header file must include a header guard to avoid name collision if
+re-included.
+
+2.6 foo._class\_
+~~~~~~~~~~~~~~~~
+
+**pandas** uses ‘type(foo)’ instead ‘foo.__class__’ as it is making the
+code more readable.
+
+For example:
+
+**Good:**
+
+::
+
+   foo = "bar"
+   type(foo)
+
+**Bad:**
+
+::
+
+   foo = "bar"
+   foo.__class__
+
+2.7 Bare Pytest Raises
+~~~~~~~~~~~~~~~~~~~~~~
+
+**Good:**
+
+::
+
+   with pytest.raise(ValueError, match="foo"):
+       # following code that raises ValueError
+
+**Bad:**
+
+::
+
+   with pytest.raise(ValueError):
+       # following code that raises ValueError
+
+3 String Formatting
+-------------------
+
+3.1 Concatenated Strings
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+3.1.1 f-strings
+^^^^^^^^^^^^^^^
+
+*pandas* uses f-strings formatting instead of ‘%’ and ‘.format()’ string
+formatters.
+
+The convention of using f-strings on a string that is concatenated over
+serveral lines, is to prefix only the lines containing the value needs
+to be interpeted.
+
+For example:
+
+**Good:**
+
+::
+
+   foo = "old_function"
+   bar = "new_function"
+
+   my_warning_message = (
+       f"Warning, {foo} is deprecated, "
+       "please use the new and way better "
+       f"{bar}"
+   )
+
+**Bad:**
+
+::
+
+   foo = "old_function"
+   bar = "new_function"
+
+   my_warning_message = (
+       f"Warning, {foo} is deprecated, "
+       f"please use the new and way better "
+       f"{bar}"
+   )
+
+3.1.2 White Spaces
+^^^^^^^^^^^^^^^^^^
+
+Putting the white space only at the end of the previous line, so there
+is no whitespace at the beggining of the concatenated string.
+
+For example:
+
+**Good:**
+
+::
+
+   example_string = (
+       "Some long concatenated string, "
+       "with good placement of the "
+       "whitespaces"
+   )
+
+**Bad:**
+
+::
+
+   example_string = (
+       "Some long concatenated string,"
+       " with bad placement of the"
+       " whitespaces"
+   )
+
+3.2 Representation function (aka ‘repr()’)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*pandas* uses ‘repr()’ instead of ‘%r’ and ‘!r’.
+
+The use of ‘repr()’ will only happend when the value is not an obvious
+string.
+
+For example:
+
+**Good:**
+
+::
+
+   value = str
+   f"Unknown recived value, got: {repr(value)}"
+
+**Bad:**
+
+::
+
+   value = str
+   f"Unknown recived type, got: '{type(value).__name__}'"
+
+4 Types

```

Comment:
OK so this is already in the contributing guideline. If we want to move here then should remove from that so as not to duplicate